jfix-armeria-facade/src/main/kotlin/ru/fix/armeria/facade/webclient/impl/BaseHttpClientBuilderImpl.kt

```diff
@@ -31,9 +30,8 @@ internal data class BaseHttpClientBuilderState(
         throw IllegalStateException("clientName must be set")
     },
     val sessionProtocol: SessionProtocol = SessionProtocol.HTTP,
-    val endpointGroupCreator: () -> Either<Lazy<EndpointGroup>, URI> = {
-        throw IllegalStateException("endpoint/endpoint_group or URI must be set")
-    },
+    val endpointGroup: EndpointGroup? = null,
```

Comment:
I think it is better to stay with `Either<EndpointGroup, URI>` declaration. Motivation is following - let's consider case:

1. builder client calls `setEndpoint(uri: URI)`
2. then he/she calls `setEndpoint(<EndpointGroup creating method>)`

Since copy of internal state is created, both `endpointGroup` and `endpoint` fields will be filled in while it is internally (in terms of simple logic) expected that only one option is chosen. 

From developer point of view you have only following options:

1. always give priority to some option (either EndpointGroup or URI). In such way you do not respect the last option chosen by builder clients (I think if client chosen option A and then option B he/she expects, that B will override A regardless of developer's internal priority choice)
2. to respect client's last choice it is required to perform `copy(endpointGroup=<some val>, endpoint=null)` (or vice versa). It would add complexity to internal builder's implementation

Either explicitly (by type) states, that some filed can have only one of defined types. So it solves both problems above.

That is why I would like to suggest the following declaration:

```suggestion
    val endpoint: Either<EndpointGroup, URI>? = null,
```

nullable is because we do not want to force clients to define endpoint